scaaml/capture/scope/scope_template.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for a scope handle. An object that can be passed as a scope to
+chipwhisperer API."""
+
+from abc import abstractmethod
+
+import numpy as np
+
+from chipwhisperer.capture.api.cwcommon import ChipWhispererCommonInterface
+
+
+class ScopeTemplate(ChipWhispererCommonInterface):
```

Comment:
Don't derive from `ChipwhispererCommonInterface`. The name is a bit misleading but if you look at the methods it has, they aren't related to the scope at all (firmware updates, etc.).

We should simply get our class, then have a `ChipwhispererScope` that derives from `ScopeTemplate` and `OpenADC`

Reply:
I am a bit lost here. I don't understand where a `ChipwhispererScope` that derives from `ScopeTemplate` and `OpenADC` comes into.

On the cw side: `OpenADC` derives from `ChipwhispererCommonInterface` [[1]](https://github.com/newaetech/chipwhisperer/blob/1a31817b2de5ac3608631ed08c09809a2d980c03/software/chipwhisperer/capture/scopes/OpenADC.py#L54). And `cw.capture_trace` argument scope [[2]](https://github.com/newaetech/chipwhisperer/blob/1a31817b2de5ac3608631ed08c09809a2d980c03/software/chipwhisperer/__init__.py#L373) has type `ScopeTypes = Union[OpenADC, CWNano]` (this is why I was afraid that mypy would consider passing any other type to be an error).

On scaaml side: We have only context managers (`CWScope, DefaultCWScope, PicoScope`) which have `.scope` property. And a single scope class acting as a `ScopeTypes` (`Pico6424E`).